instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb

```diff
@@ -205,123 +205,226 @@ def finish(name, id, payload)
-          _(last_span.attributes['extra']).must_equal('context')
-        end
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal(notification_name)
+        _(last_span.attributes['extra']).must_equal('context')
       end
 
-      describe 'when using a unstable formatter' do
-        it 'defaults to the notification name' do
-          allow(OpenTelemetry).to receive(:handle_error).with(exception: RuntimeError, message: String)
+      it 'finishes spans even when complex subscribers blow up' do
+        ActiveSupport::Notifications.subscribe(notification_name, CrashingEndSubscriber.new)
+        OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
 
-          OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name, nil, nil, span_name_formatter: ->(_) { raise 'boom' })
+        expect do
           ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+        end.must_raise RuntimeError
 
-          _(last_span).wont_be_nil
-          _(last_span.name).must_equal(notification_name)
-          _(last_span.attributes['extra']).must_equal('context')
-        end
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal(notification_name)
+        _(last_span.attributes['extra']).must_equal('context')
       end
-    end
 
-    it 'finishes spans even when block subscribers blow up' do
-      ActiveSupport::Notifications.subscribe(notification_name) { raise 'boom' }
-      OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
+      it 'supports unsubscribe' do
+        obj = OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
+        ActiveSupport::Notifications.unsubscribe(obj)
 
-      expect do
         ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
-      end.must_raise RuntimeError
 
-      _(last_span).wont_be_nil
-      _(last_span.name).must_equal(notification_name)
-      _(last_span.attributes['extra']).must_equal('context')
+        _(obj.class).must_equal(ActiveSupport::Notifications::Fanout::Subscribers::Evented)
+        _(last_span).must_be_nil
+      end
+
+      it 'supports setting the span kind' do
+        OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, 'bar.foo', nil, [], kind: :client)
+        ActiveSupport::Notifications.instrument('bar.foo', extra: 'context')
+
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal('bar.foo')
+        _(last_span.attributes['extra']).must_equal('context')
+        _(last_span.kind).must_equal(:client)
+      end
     end
 
-    it 'finishes spans even when complex subscribers blow up' do
-      ActiveSupport::Notifications.subscribe(notification_name, CrashingEndSubscriber.new)
-      OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
+    describe 'when subscribing to events matching a regular expression' do
```

Comment:
This context contains the same tests as above, but exercising the pattern subscription.